packages/react/src/components/slot/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+
+export type Props<Name extends string> = {
+  children: React.ReactNode;
+  name: Name; // name of the slot the component is being rendered into
+  asChild?: boolean; // pass props to the immediate child
+};
+
+// TODO consider using asChild prop as an alternative to render props = passes props to the immediate child
+
+// add render props here
+
+/**
+ * TODO document this component
+ * - issue: it is not typesafe, it is possible to render a slot with a name that does not exist
+ * @returns
+ */
+export function Slot<N extends string>({ children, name, asChild }: Props<N>) {
```

Comment:
I couldn't get TS to help - I wanted to see if we could somehow annotate the components with slots on the type level, so TS gives you instant feedback when passing down a slot with an incorrect name.